### DISCUSSION TITLE
Library for Tree/Graph visualization

### DISCUSSION THREAD

**Discussion by PSNAppz** [DedSecInside/TorBoT#256] [Ideas]
Since the release of TorBot v1, we were using the `ete3` library which served the purpose. But unfortunately, due to missing support for `PyQt5` and `python3.6`^ we had to remove the visualization module at least until an update for `ete3` is released. 

But since there have been no updates for over a year, we are looking for better alternatives to the visualization module. I'm opening this issue so that the community can decide on the library.

**Comment by sebastian8456** [comment]
Check out Matplotlib, it is a popular data visualization library.

**Comment by PSNAppz** [comment]
As far as I know, matplotlib doesn't support tree/graph visualization. (see: https://matplotlib.org/stable/plot_types/index.html)

**Comment by PSNAppz** [comment]
Either we can use [Plotly](https://plotly.com/python/tree-plots/) which again uses 3rd party Graphic Libraries or we can use [D3.js](https://www.google.com/search?client=safari&rls=en&q=d3js&ie=UTF-8&oe=UTF-8) but we'll have to render the result in a webpage (which is fine in my opinion).
